### ISSUE TITLE
Regression in speed tests

### ISSUE THREAD

**Issue by PiotrSikora** [envoyproxy/envoy-wasm#602]
Tracking issue, results to follow.

cc @mandarjog @jplevyak @duderino @Aakash2017

**Comment by PiotrSikora** [comment]
From `a9dabc9e722d78214cbf4d5961d68ffe737d6e9c`:
```
-------------------------------------------------------------------------------------------------------------------
Benchmark                                                                         Time             CPU   Iterations
-------------------------------------------------------------------------------------------------------------------
BM_WasmSimpleCallSpeedTest/V8SpeedTest_empty                                   85.1 ns         85.1 ns      7471294
BM_WasmSimpleCallSpeedTest/NullSpeedTest_empty                                 21.3 ns         21.2 ns     32869397
BM_WasmSimpleCallSpeedTest/V8SpeedTest_get_current_time                         285 ns          285 ns      2341191
BM_WasmSimpleCallSpeedTest/NullSpeedTest_get_current_time                       108 ns          108 ns      6680107
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string                             214 ns          214 ns      2999100
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string                          33.6 ns         33.6 ns     20782370
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string1000                      123592 ns       123481 ns         4929
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string1000                     12061 ns        12059 ns        57880
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string_check_compiler              396 ns          396 ns      1725302
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string_check_compiler           54.8 ns         54.8 ns     13113224
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string_check_compiler1000       222189 ns       222008 ns         3039
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string_check_compiler1000      22330 ns        22328 ns        31398
BM_WasmSimpleCallSpeedTest/V8SpeedTest_large_string                            1188 ns         1187 ns       518886
BM_WasmSimpleCallSpeedTest/NullSpeedTest_large_string                           105 ns          105 ns      6667727
BM_WasmSimpleCallSpeedTest/V8SpeedTest_large_string1000                     1046548 ns      1045742 ns          591
BM_WasmSimpleCallSpeedTest/NullSpeedTest_large_string1000                     82123 ns        82115 ns         8520
BM_WasmSimpleCallSpeedTest/V8SpeedTest_get_property                             464 ns          463 ns      1210388
BM_WasmSimpleCallSpeedTest/NullSpeedTest_get_property                           132 ns          132 ns      5287100
BM_WasmSimpleCallSpeedTest/V8SpeedTest_grpc_service                             517 ns          516 ns      1335983
BM_WasmSimpleCallSpeedTest/NullSpeedTest_grpc_service                           114 ns          114 ns      6063942
BM_WasmSimpleCallSpeedTest/V8SpeedTest_grpc_service1000                      405235 ns       404867 ns         1601
BM_WasmSimpleCallSpeedTest/NullSpeedTest_grpc_service1000                     88094 ns        88085 ns         7928
BM_WasmSimpleCallSpeedTest/V8SpeedTest_modify_metadata                          954 ns          953 ns       634406
BM_WasmSimpleCallSpeedTest/NullSpeedTest_modify_metadata                        179 ns          179 ns      3912613
BM_WasmSimpleCallSpeedTest/V8SpeedTest_modify_metadata1000                   802532 ns       801760 ns          733
BM_WasmSimpleCallSpeedTest/NullSpeedTest_modify_metadata1000                 152988 ns       152973 ns         4578
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize                        179541 ns       179457 ns         3463
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize                       29432 ns        29429 ns        23770
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize_arena                  191952 ns       191841 ns         3199
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize_arena                 30333 ns        30329 ns        23986
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize                        5561 ns         5556 ns        95231
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize                     12267 ns        12266 ns        57090
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize_arena                  5716 ns         5710 ns        89885
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize_arena               12590 ns        12589 ns        58035
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize_empty                  6388 ns         6383 ns        87599
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize_empty                 874 ns          874 ns       790081
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize_deserialize            259103 ns       258909 ns         2513
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize_deserialize           42696 ns        42692 ns        16373
BM_WasmSimpleCallSpeedTest/V8SpeedTest_convert_to_filter_state               271312 ns       271046 ns         2378
BM_WasmSimpleCallSpeedTest/NullSpeedTest_convert_to_filter_state              38078 ns        38074 ns        18305
```

**Comment by PiotrSikora** [comment]
From `a9dabc9e722d78214cbf4d5961d68ffe737d6e9c` (using V8 reverted to 8.3.110.9):
```
-------------------------------------------------------------------------------------------------------------------
Benchmark                                                                         Time             CPU   Iterations
-------------------------------------------------------------------------------------------------------------------
BM_WasmSimpleCallSpeedTest/V8SpeedTest_empty                                   88.9 ns         88.8 ns      7266079
BM_WasmSimpleCallSpeedTest/NullSpeedTest_empty                                 22.2 ns         22.2 ns     31624117
BM_WasmSimpleCallSpeedTest/V8SpeedTest_get_current_time                         282 ns          282 ns      2481681
BM_WasmSimpleCallSpeedTest/NullSpeedTest_get_current_time                       105 ns          105 ns      6647490
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string                             226 ns          226 ns      2932471
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string                          32.9 ns         32.9 ns     21303771
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string1000                      121803 ns       121712 ns         5057
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string1000                     12298 ns        12296 ns        56919
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string_check_compiler              417 ns          417 ns      1664366
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string_check_compiler           53.4 ns         53.4 ns     13558119
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string_check_compiler1000       214488 ns       214348 ns         3088
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string_check_compiler1000      21878 ns        21876 ns        31518
BM_WasmSimpleCallSpeedTest/V8SpeedTest_large_string                            1178 ns         1177 ns       542065
BM_WasmSimpleCallSpeedTest/NullSpeedTest_large_string                           105 ns          105 ns      6494627
BM_WasmSimpleCallSpeedTest/V8SpeedTest_large_string1000                     1064307 ns      1063480 ns          605
BM_WasmSimpleCallSpeedTest/NullSpeedTest_large_string1000                     84677 ns        84667 ns         8166
BM_WasmSimpleCallSpeedTest/V8SpeedTest_get_property                             437 ns          437 ns      1557825
BM_WasmSimpleCallSpeedTest/NullSpeedTest_get_property                           132 ns          132 ns      5318519
BM_WasmSimpleCallSpeedTest/V8SpeedTest_grpc_service                             522 ns          522 ns      1264117
BM_WasmSimpleCallSpeedTest/NullSpeedTest_grpc_service                           115 ns          115 ns      6079598
BM_WasmSimpleCallSpeedTest/V8SpeedTest_grpc_service1000                      441783 ns       441392 ns         1651
BM_WasmSimpleCallSpeedTest/NullSpeedTest_grpc_service1000                     89015 ns        89006 ns         7853
BM_WasmSimpleCallSpeedTest/V8SpeedTest_modify_metadata                          937 ns          936 ns       708861
BM_WasmSimpleCallSpeedTest/NullSpeedTest_modify_metadata                        183 ns          183 ns      3757659
BM_WasmSimpleCallSpeedTest/V8SpeedTest_modify_metadata1000                   800211 ns       799620 ns          811
BM_WasmSimpleCallSpeedTest/NullSpeedTest_modify_metadata1000                 152555 ns       152541 ns         4577
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize                        178988 ns       178903 ns         3222
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize                       29268 ns        29266 ns        23909
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize_arena                  196606 ns       196520 ns         3147
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize_arena                 30537 ns        30534 ns        24493
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize                        5533 ns         5528 ns        91200
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize                     12245 ns        12244 ns        57584
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize_arena                  5934 ns         5930 ns       100105
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize_arena               12183 ns        12182 ns        57188
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize_empty                  5932 ns         5928 ns        94037
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize_empty                 859 ns          859 ns       810393
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize_deserialize            256752 ns       256567 ns         2422
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize_deserialize           42532 ns        42527 ns        16503
BM_WasmSimpleCallSpeedTest/V8SpeedTest_convert_to_filter_state               265166 ns       265011 ns         2376
BM_WasmSimpleCallSpeedTest/NullSpeedTest_convert_to_filter_state              37433 ns        37429 ns        18718
```

**Comment by PiotrSikora** [comment]
From `a9dabc9e722d78214cbf4d5961d68ffe737d6e9c` (using V8 reverted to 8.1.307.28):
```
-------------------------------------------------------------------------------------------------------------------
Benchmark                                                                         Time             CPU   Iterations
-------------------------------------------------------------------------------------------------------------------
BM_WasmSimpleCallSpeedTest/V8SpeedTest_empty                                   90.1 ns         90.0 ns      7571261
BM_WasmSimpleCallSpeedTest/NullSpeedTest_empty                                 21.5 ns         21.5 ns     32877542
BM_WasmSimpleCallSpeedTest/V8SpeedTest_get_current_time                         263 ns          263 ns      2452525
BM_WasmSimpleCallSpeedTest/NullSpeedTest_get_current_time                       104 ns          104 ns      6750176
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string                             216 ns          216 ns      2987770
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string                          33.9 ns         33.9 ns     20765484
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string1000                      125694 ns       125606 ns         5164
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string1000                     12152 ns        12151 ns        56516
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string_check_compiler              403 ns          403 ns      1719895
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string_check_compiler           52.3 ns         52.3 ns     13722077
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string_check_compiler1000       213570 ns       213440 ns         3027
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string_check_compiler1000      22757 ns        22754 ns        30814
BM_WasmSimpleCallSpeedTest/V8SpeedTest_large_string                            1191 ns         1190 ns       534944
BM_WasmSimpleCallSpeedTest/NullSpeedTest_large_string                           104 ns          104 ns      6749620
BM_WasmSimpleCallSpeedTest/V8SpeedTest_large_string1000                     1071094 ns      1070336 ns          607
BM_WasmSimpleCallSpeedTest/NullSpeedTest_large_string1000                     80033 ns        80025 ns         8773
BM_WasmSimpleCallSpeedTest/V8SpeedTest_get_property                             451 ns          451 ns      1453421
BM_WasmSimpleCallSpeedTest/NullSpeedTest_get_property                           130 ns          130 ns      5332521
BM_WasmSimpleCallSpeedTest/V8SpeedTest_grpc_service                             553 ns          553 ns       946173
BM_WasmSimpleCallSpeedTest/NullSpeedTest_grpc_service                           115 ns          114 ns      6079291
BM_WasmSimpleCallSpeedTest/V8SpeedTest_grpc_service1000                      423089 ns       422776 ns         1668
BM_WasmSimpleCallSpeedTest/NullSpeedTest_grpc_service1000                     89858 ns        89849 ns         7745
BM_WasmSimpleCallSpeedTest/V8SpeedTest_modify_metadata                          928 ns          928 ns       704314
BM_WasmSimpleCallSpeedTest/NullSpeedTest_modify_metadata                        187 ns          187 ns      3761326
BM_WasmSimpleCallSpeedTest/V8SpeedTest_modify_metadata1000                   806227 ns       805524 ns          826
BM_WasmSimpleCallSpeedTest/NullSpeedTest_modify_metadata1000                 157086 ns       157070 ns         4478
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize                        185319 ns       185253 ns         3110
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize                       29758 ns        29755 ns        23620
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize_arena                  197165 ns       196986 ns         3202
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize_arena                 31078 ns        31074 ns        23880
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize                        5516 ns         5510 ns        96843
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize                     12071 ns        12069 ns        57940
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize_arena                  5642 ns         5635 ns        96950
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize_arena               11978 ns        11976 ns        58009
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize_empty                  5897 ns         5891 ns        94872
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize_empty                 863 ns          863 ns       819098
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize_deserialize            264294 ns       264001 ns         2397
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize_deserialize           42630 ns        42626 ns        16292
BM_WasmSimpleCallSpeedTest/V8SpeedTest_convert_to_filter_state               273061 ns       272907 ns         2348
BM_WasmSimpleCallSpeedTest/NullSpeedTest_convert_to_filter_state              37918 ns        37914 ns        18396
```

**Comment by PiotrSikora** [comment]
It's not V8, since it looks that all the tests (other than `small_string_check_compiler1000 `?) have generally improved.

**Comment by PiotrSikora** [comment]
OK, re-enabling LTO (https://github.com/proxy-wasm/proxy-wasm-cpp-sdk/pull/44) did the trick:
```
-------------------------------------------------------------------------------------------------------------------
Benchmark                                                                         Time             CPU   Iterations
-------------------------------------------------------------------------------------------------------------------
BM_WasmSimpleCallSpeedTest/V8SpeedTest_empty                                   88.5 ns         88.4 ns      6855570
BM_WasmSimpleCallSpeedTest/NullSpeedTest_empty                                 21.3 ns         21.3 ns     32826442
BM_WasmSimpleCallSpeedTest/V8SpeedTest_get_current_time                         285 ns          285 ns      2564993
BM_WasmSimpleCallSpeedTest/NullSpeedTest_get_current_time                       104 ns          104 ns      6694911
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string                             131 ns          131 ns      4424660
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string                          33.7 ns         33.7 ns     20826395
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string1000                       40923 ns        40902 ns        16130
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string1000                     12088 ns        12087 ns        58078
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string_check_compiler              181 ns          180 ns      3539921
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string_check_compiler           51.7 ns         51.7 ns     13664469
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string_check_compiler1000        72298 ns        72259 ns         9219
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string_check_compiler1000      22274 ns        22272 ns        31501
BM_WasmSimpleCallSpeedTest/V8SpeedTest_large_string                            1052 ns         1051 ns       515705
BM_WasmSimpleCallSpeedTest/NullSpeedTest_large_string                           107 ns          107 ns      6620549
BM_WasmSimpleCallSpeedTest/V8SpeedTest_large_string1000                      852292 ns       851648 ns          755
BM_WasmSimpleCallSpeedTest/NullSpeedTest_large_string1000                     84182 ns        84173 ns         8254
BM_WasmSimpleCallSpeedTest/V8SpeedTest_get_property                             453 ns          453 ns      1191695
BM_WasmSimpleCallSpeedTest/NullSpeedTest_get_property                           132 ns          132 ns      5348841
BM_WasmSimpleCallSpeedTest/V8SpeedTest_grpc_service                             361 ns          361 ns      1875879
BM_WasmSimpleCallSpeedTest/NullSpeedTest_grpc_service                           114 ns          114 ns      6163003
BM_WasmSimpleCallSpeedTest/V8SpeedTest_grpc_service1000                      232765 ns       232589 ns         2675
BM_WasmSimpleCallSpeedTest/NullSpeedTest_grpc_service1000                     88414 ns        88406 ns         7920
BM_WasmSimpleCallSpeedTest/V8SpeedTest_modify_metadata                         1029 ns         1028 ns       595400
BM_WasmSimpleCallSpeedTest/NullSpeedTest_modify_metadata                        178 ns          178 ns      3914789
BM_WasmSimpleCallSpeedTest/V8SpeedTest_modify_metadata1000                   883704 ns       883137 ns          662
BM_WasmSimpleCallSpeedTest/NullSpeedTest_modify_metadata1000                 153067 ns       153052 ns         4568
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize                         97456 ns        97408 ns         6842
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize                       29295 ns        29292 ns        23879
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize_arena                  116366 ns       116285 ns         4503
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize_arena                 29082 ns        29079 ns        24036
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize                        2887 ns         2885 ns       233640
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize                     12299 ns        12298 ns        57156
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize_arena                  2691 ns         2690 ns       235299
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize_arena               12119 ns        12118 ns        57506
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize_empty                  2795 ns         2792 ns       234550
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize_empty                 838 ns          838 ns       839364
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize_deserialize            155315 ns       155216 ns         3608
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize_deserialize           42488 ns        42484 ns        16441
BM_WasmSimpleCallSpeedTest/V8SpeedTest_convert_to_filter_state               135826 ns       135744 ns         3922
BM_WasmSimpleCallSpeedTest/NullSpeedTest_convert_to_filter_state              59503 ns        59496 ns        16930
```

**Comment by PiotrSikora** [comment]
With LTO (proxy-wasm/proxy-wasm-cpp-sdk#44) and `--llvm-lto 3`:
```
-------------------------------------------------------------------------------------------------------------------
Benchmark                                                                         Time             CPU   Iterations
-------------------------------------------------------------------------------------------------------------------
BM_WasmSimpleCallSpeedTest/V8SpeedTest_empty                                    102 ns          102 ns      5955889
BM_WasmSimpleCallSpeedTest/NullSpeedTest_empty                                 26.2 ns         26.2 ns     26784239
BM_WasmSimpleCallSpeedTest/V8SpeedTest_get_current_time                         275 ns          275 ns      2381960
BM_WasmSimpleCallSpeedTest/NullSpeedTest_get_current_time                       109 ns          109 ns      6463193
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string                             136 ns          136 ns      4438922
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string                          36.9 ns         36.9 ns     18996423
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string1000                       42033 ns        42007 ns        16090
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string1000                     12065 ns        12064 ns        57933
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string_check_compiler              183 ns          183 ns      3305627
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string_check_compiler           55.8 ns         55.8 ns     12771851
BM_WasmSimpleCallSpeedTest/V8SpeedTest_small_string_check_compiler1000        70995 ns        70934 ns         9510
BM_WasmSimpleCallSpeedTest/NullSpeedTest_small_string_check_compiler1000      22220 ns        22218 ns        31414
BM_WasmSimpleCallSpeedTest/V8SpeedTest_large_string                             958 ns          957 ns       665401
BM_WasmSimpleCallSpeedTest/NullSpeedTest_large_string                           108 ns          108 ns      6459128
BM_WasmSimpleCallSpeedTest/V8SpeedTest_large_string1000                      895316 ns       894747 ns          778
BM_WasmSimpleCallSpeedTest/NullSpeedTest_large_string1000                     81146 ns        81138 ns         8616
BM_WasmSimpleCallSpeedTest/V8SpeedTest_get_property                             477 ns          477 ns      1172609
BM_WasmSimpleCallSpeedTest/NullSpeedTest_get_property                           133 ns          133 ns      5216334
BM_WasmSimpleCallSpeedTest/V8SpeedTest_grpc_service                             342 ns          342 ns      1867000
BM_WasmSimpleCallSpeedTest/NullSpeedTest_grpc_service                           123 ns          123 ns      5876749
BM_WasmSimpleCallSpeedTest/V8SpeedTest_grpc_service1000                      260565 ns       260363 ns         2574
BM_WasmSimpleCallSpeedTest/NullSpeedTest_grpc_service1000                     95422 ns        95413 ns         7049
BM_WasmSimpleCallSpeedTest/V8SpeedTest_modify_metadata                         1074 ns         1073 ns       533145
BM_WasmSimpleCallSpeedTest/NullSpeedTest_modify_metadata                        199 ns          199 ns      3465477
BM_WasmSimpleCallSpeedTest/V8SpeedTest_modify_metadata1000                   973648 ns       972763 ns          597
BM_WasmSimpleCallSpeedTest/NullSpeedTest_modify_metadata1000                 153008 ns       152992 ns         4575
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize                        101050 ns       101010 ns         5061
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize                       29103 ns        29100 ns        23894
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize_arena                  115502 ns       115415 ns         4479
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize_arena                 29168 ns        29165 ns        23952
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize                        2641 ns         2639 ns       250186
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize                     12307 ns        12306 ns        57128
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize_arena                  2617 ns         2615 ns       245580
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize_arena               12172 ns        12171 ns        57160
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_deserialize_empty                  2885 ns         2884 ns       226592
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_deserialize_empty                 859 ns          859 ns       811943
BM_WasmSimpleCallSpeedTest/V8SpeedTest_json_serialize_deserialize            152256 ns       152124 ns         3553
BM_WasmSimpleCallSpeedTest/NullSpeedTest_json_serialize_deserialize           42426 ns        42422 ns        16313
BM_WasmSimpleCallSpeedTest/V8SpeedTest_convert_to_filter_state               133539 ns       133370 ns         4025
BM_WasmSimpleCallSpeedTest/NullSpeedTest_convert_to_filter_state              37579 ns        37576 ns        18661
```

**Comment by PiotrSikora** [comment]
Fixed in proxy-wasm/proxy-wasm-cpp-sdk#44.
